builder/vmware/common/hw_config.go

```diff
@@ -318,30 +315,30 @@ func (c *HWConfig) ReadSerial() (*SerialUnion, error) {
 			res.Yield = strings.ToUpper(comp[1])
 		}
 		if res.Yield != "TRUE" && res.Yield != "FALSE" {
-			return nil, fmt.Errorf("Unexpected format for yield in serial port file: %s -> %s", c.Serial, res.Yield)
+			return nil, fmt.Errorf("unexpected format for yield in serial port file: %s -> %s", c.Serial, res.Yield)
 		}
 
 		return &SerialUnion{Union: res, File: res}, nil
 
 	case "DEVICE":
 		comp := strings.Split(formatOptions, ",")
 		if len(comp) > 2 {
-			return nil, fmt.Errorf("Unexpected format for serial port device: %s", c.Serial)
+			return nil, fmt.Errorf("unexpected format for serial port device: %s", c.Serial)
 		}
 		res := new(SerialConfigDevice)
 		// set serial port defaults
 		res.Devicename = defaultSerialPort
 		res.Yield = "FALSE"
 		// Read actual values from component, if set.
 		if len(comp) == 1 {
-			filepath.FromSlash(comp[0])
+			comp[0] = filepath.FromSlash(comp[0])
```

Comment:
Nice catch, however is `comp[0]` used later? It doesn't seem to be the case

Reply:
Doesn't appear to be used later.

Reply:
builder/vmware/common/hw_config.go:334:4: FromSlash doesn't have side effects and its return value is ignored (SA4017)

Probably better to only have:

```go
if len(comp) == 2 {
    res.Devicename = filepath.FromSlash(comp[0])
    res.Yield = strings.ToUpper(comp[1])
}
```

Reply:
Lemme know what you think. Otherwise we could revert to as it was.